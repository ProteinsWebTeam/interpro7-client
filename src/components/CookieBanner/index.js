// @flow
import React, {Component} from 'react';

import AnimatedEntry from 'components/AnimatedEntry';

import foundation from 'styles/foundation';

class CookieBanner extends Component {
  /* ::
    state: {display: ?bool}
    _banner: ?any
  */
  componentWillMount() {
    if (document) {
      this.setState({
        // If cookies are already accepted, display -> false
        display: !(document.cookie.match(/cookies-accepted=(true)/i) || [])[1],
      });
    }
  }

  handleClick = () => {
    document.cookie = 'cookies-accepted=true';
    if (this._banner && this._banner.animate) {
      this._banner
        .animate([
          {transform: 'translateY(0)', opacity: 1},
          {transform: 'translateY(100%)', opacity: 0.5},
        ], {duration: 300, fill: 'forwards', easing: 'ease-in'})
        .onfinish = () => this.setState({display: false});
    } else {
      this.setState({display: false});
    }
  };

  render() {
    const {display = false} = this.state;
    if (!display) return null;
    return (
      <AnimatedEntry
        animateSelf={true}
        keyframes={{
          opacity: [0, 1],
          transform: ['translateY(100%)', 'translateY(0)'],
        }}
        element="div"
        style={{
          position: 'fixed',
          bottom: 0,
          width: '100vw',
          background: 'rgba(0, 0, 0, 0.85)',
          color: '#eee',
          padding: '0.75em',
          zIndex: 101,
        }}
      >
        <div className={foundation('row')}>
        <span style={{marginRight: '2em', flex: 1}} >
          This website uses cookies.
          By continuing to browse this site,
          you are agreeing to the use of our site cookies.
          To find out more, see our{' '}
          <a
            target="_blank"
            href="https://www.ebi.ac.uk/about/terms-of-use"
          >
            Terms of Use
          </a>.
        </span>
        <button
          style={{fontWeight: 'bold', padding: '0.5em'}}
          onClick={this.handleClick}
        >
          Ã—
        </button>
        </div>
      </AnimatedEntry>
    );
  }
}

export default CookieBanner;
