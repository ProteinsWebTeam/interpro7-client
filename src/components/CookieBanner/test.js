// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import CookieBanner from '.';

const renderer = new ShallowRenderer();

describe('<CookieBanner />', () => {
  test('should render', () => {
    renderer.render(<CookieBanner />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
