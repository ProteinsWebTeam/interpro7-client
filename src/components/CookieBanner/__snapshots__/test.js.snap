// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CookieBanner /> should render 1`] = `
<Memo(Connect(AnimatedEntry))
  animateSelf={true}
  element="div"
  keyframes={
    Object {
      "opacity": Array [
        0,
        1,
      ],
      "transform": Array [
        "translateY(100%)",
        "translateY(0)",
      ],
    }
  }
  style={
    Object {
      "background": "rgb(112, 115, 114)",
      "bottom": 0,
      "color": "#eee",
      "padding": "0.75em",
      "position": "fixed",
      "width": "100vw",
      "zIndex": 101,
    }
  }
>
  <div
    className="row"
    style={
      Object {
        "alignItems": "baseline",
        "display": "flex",
      }
    }
  >
    <div
      className="columns medium-8 large-9 white-color"
    >
      This website requires cookies, and the limited processing of your personal data in order to function. By using the site you are agreeing to this as outlined in our
       
      <Memo(Connect(Link))
        href="https://www.ebi.ac.uk/data-protection/privacy-notice/embl-ebi-public-website"
        style={
          Object {
            "color": "#f8f8f8",
          }
        }
        target="_blank"
      >
        Privacy Notice
      </Memo(Connect(Link))>
       and 
      <Memo(Connect(Link))
        href="https://www.ebi.ac.uk/about/terms-of-use"
        style={
          Object {
            "color": "#f8f8f8",
          }
        }
        target="_blank"
      >
        Terms of Use
      </Memo(Connect(Link))>
      .
    </div>
    <div
      className="columns medium-4 large-3 text-right white-color"
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        I agree, dismiss this banner
      </button>
    </div>
  </div>
</Memo(Connect(AnimatedEntry))>
`;
